### PR TITLE
[bitnami/nginx] Add ability to use ingress

### PR DESCRIPTION
**Pull request by RothAndrew**
**Description of the change**

This PR adds the ability to use an Ingress resource with the Nginx chart. Ingress resources are very popularly used due to the prevalence of ingress controllers like the `nginx-ingress-controller`.

**Benefits**

Users of this chart will now be able to configure ingress, rather than having to rely on the service with type LoadBalancer

**Possible drawbacks**

None. The ingress by default is disabled.

**Applicable issues**

N/A

**Additional information**

The new Ingress resource is identical to the one that gets created when you create a new helm chart uisng `helm create`. The values in values.yaml for the ingress are likewise identical to the ones that are generated using `helm create`

I wasn't sure how you guys handle incrementing the chart version. I went ahead and incremented to v3.2.0. Feel free to pull that out, or ask me to do it, if that isn't what you want.

Note: I just realized I forgot to add new rows to the table at the bottom of the README. I'll do that now and push another commit to this PR.

### CODE REVIEW

File: bitnami/nginx/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ $fullName }}
+  labels:
+    app: {{ $fullName }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+{{- with .Values.ingress.annotations }}
```
Comment (juan131) [inline bitnami/nginx/templates/ingress.yaml]: On other charts we included the parameter `ingress.certManager` that allows setting the corresponding annotations for people using cert-manager to handle the certificates (see https://github.com/bitnami/charts/blob/master/upstreamed/wordpress/values.yaml#L214):

```yaml
    {{- if .Values.ingress.certManager }}
    kubernetes.io/tls-acme: "true"
    {{- end }}
```

Here you have an example: https://github.com/bitnami/charts/blob/master/upstreamed/wordpress/templates/ingress.yaml#L11

File: bitnami/nginx/templates/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+{{- if .Values.ingress.tls }}
+  tls:
+  {{- range .Values.ingress.tls }}
+    - hosts:
+      {{- range .hosts }}
+        - {{ . | quote }}
+      {{- end }}
+      secretName: {{ .secretName }}
+  {{- end }}
+{{- end }}
```
Comment (juan131) [inline bitnami/nginx/templates/ingress.yaml]: You can use sth like:

```yaml
  {{- if .Values.ingress.tls }}
  tls:
{{ toYaml .Values.ingress.tls | indent 4 }}
  {{- end }}
```
And update the values.yaml:

```
  ## The tls configuration for the ingress
  ## see: https://kubernetes.io/docs/concepts/services-networking/ingress/#tls
  ## 
  tls:
  - hosts:
      - nginx.local
    secretName: nginx.local-tls
```

File: bitnami/nginx/templates/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  rules:
+  {{- range .Values.ingress.hosts }}
+    - host: {{ . | quote }}
```
Comment (juan131) [inline bitnami/nginx/templates/ingress.yaml]: Convert the parameter `ingress.hosts` into:
```
hosts:
    - name: nginx.local
       path: /
```

And then, use `{{ .name }}`

File: bitnami/nginx/templates/ingress.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      http:
+        paths:
+          - path: {{ $ingressPath }}
```
Comment (juan131) [inline bitnami/nginx/templates/ingress.yaml]: Convert the parameter ingress.hosts into:
```
hosts:
    - name: nginx.local
       path: /
```
And then, use {{ .path }}

File: bitnami/nginx/values.yaml [hunk 5]
```diff
@@ -44,6 +44,19 @@ service:
...
+    # kubernetes.io/ingress.class: nginx
+    # kubernetes.io/tls-acme: "true"
+  path: /
+  hosts:
+    - chart-example.local
+  tls: []
+  #  - secretName: chart-example-tls
+  #    hosts:
+  #      - chart-example.local
+
```
Comment (juan131) [inline bitnami/nginx/values.yaml]: Try to follow the standard we used on other charts: https://github.com/bitnami/charts/blob/master/upstreamed/wordpress/values.yaml#L206

**Comment by RothAndrew** [comment]
Hi @juan131, thanks for the review. I'll address the changes you requested and get another commit pushed this weekend.

**Comment by juan131** [comment]
Great @RothAndrew 

Please ping when your changes are ready to be reviewed!

File: bitnami/nginx/templates/ingress.yaml (hunk 1 above)
Comment (RothAndrew) [reply bitnami/nginx/templates/ingress.yaml]: done

File: bitnami/nginx/templates/ingress.yaml (hunk 2 above)
Comment (RothAndrew) [reply bitnami/nginx/templates/ingress.yaml]: done

File: bitnami/nginx/templates/ingress.yaml (hunk 3 above)
Comment (RothAndrew) [reply bitnami/nginx/templates/ingress.yaml]: done

File: bitnami/nginx/templates/ingress.yaml (hunk 4 above)
Comment (RothAndrew) [reply bitnami/nginx/templates/ingress.yaml]: done

File: bitnami/nginx/values.yaml (hunk 5 above)
Comment (RothAndrew) [reply bitnami/nginx/values.yaml]: done. Followed standard set in wordpress chart

**Comment by RothAndrew** [comment]
@juan131 Ready for re-review

File: bitnami/nginx/README.md [hunk 6]
```diff
@@ -42,31 +42,41 @@ The command removes all the Kubernetes components associated with the chart and
...
+| `service.externalTrafficPolicy`  | Enable client source IP preservation             | `Cluster`                                                    |
+| `service.loadBalancerIP`         | LoadBalancer service IP address                  | `""`                                                         |
+| `service.annotations`            | Service annotations                              | `{}`                                                         |
+| `ingress.enabled`                | Enable ingress controller resource               | `false`                                                      |
+| `ingress.certManager`            | Add annotations for cert-manager                 | `false`                                                      |
+| `ingress.annotations`            | Ingress annotations                              | `[]`                                                         |
+| `ingress.hosts[0].name`          | Hostname to your Wordpress installation          | `nginx.local`                                                |
```
Comment (juan131) [inline bitnami/nginx/README.md]: Typo: `WordPress` -> `NGINX`

**Comment by RothAndrew** [comment]
Oops! I was sure I got all of those..

:ping: @juan131 ready for re-re-review.

**Pull request merged**